### PR TITLE
Feature/database rewrite/m habermas

### PR DESCRIPTION
**Pull request by MichaelHabermas**
Major DB update to address structural concerns.

[ DB Designer Schema Link](https://dbdesigner.page.link/TkqhaqDpvCjd4dDJ8)

## Change Status

- Completed, ready to review and merge

## Checklist

- [x] Endpoint tested with Postman, or Unit test.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] There are no merge conflicts

### CODE REVIEW

**Review by ahmedseragcodes — APPROVED** [review]
Changes make sense contextually / logically with other code and according to plan, approved

**Review by blackenjoy27 — APPROVED** [review]
I think the change is good. The new functions Michael added and the one that he deleted make senses. I didn't find anything bad. The change he made to the database table make more sense than the previous one. Nice work!

File: data/seeds/015-recipients.js [hunk 1]
```diff
@@ -8,21 +8,34 @@ const {
 } = require('../seedHelpers');
 
 const recipients = fakeRecipientIds.map((id) => {
+  const genID = getRand(19);
```
Comment (santorap) [inline data/seeds/015-recipients.js:11]: In the future, these hardcoded values for each of these columns may need to be dynamically generated. (Or, ideally, we get a much better understanding of the distribution of the actual data for these values and reflect that when seeding the DB.)

File: data/seeds/018-service_entries.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    service_date: faker.date.past(3),
+    service_time: faker.time.recent('abbr'),
+    service_duration: null,
+    service_value: getRandWithZero(300),
+    service_quantity: null,
+    service_entry_notes: '',
+    // service_entry_data: {
+    //   custom: fakeServiceEntryTypeCustomData(serviceTypeNumToUse - 1), // In this case using '-1' since the fake service creator is zero indexed
+    // },
+    service_entry_data: {}, // keeping above for now, may get rid of it
```
Comment (santorap) [inline data/seeds/018-service_entries.js:35]: Will likely need to build out the structure to allow for multiple (non-primary) providers and/or recipients into this jsonb object at a later date.

**Review by santorap — APPROVED** [review]
This was a massive effort! I reviewed the changes as best as I could, but at some point we need to make a judgement call. I think we need to push forward! Great job!!

**Pull request merged**